main/views.py

```diff
@@ -13,7 +15,24 @@ def index(request):
 
 @login_required()
 def homepage(request):
-    return render(request, 'main/homepage.html')
+    if DogOwner.objects.filter(user=request.user).exists():
+        dog_owner = DogOwner.objects.filter(user=request.user).first()
+        profile_picture_url = dog_owner.dog_picture_url
+        username = dog_owner.first_name + dog_owner.last_name
+        dog_name = dog_owner.dog_name
+    else:
+        daycare = DayCare.objects.filter(user=request.user).first()
+        profile_picture_url = Image.objects.filter(daycare_id=daycare).first()
+        username = daycare.name
+        dog_name = None
+
+    context = {
+        'profile_picture_url': profile_picture_url,
+        'username': username,
+        'dog_name': dog_name,
+    }
+
+    return render(request, 'main/homepage.html', context)
```

Comment:
Why are you returning the first dog-owner or daycare object here? Is not it better that we will check if the user exists, and then call for a function of a homepage by the type of the object?
 If it's a dog owner then `dogwner_homepage` and in the same way for daycare? As we did from the beginning and you can see it [here](#63)

Reply:
Hi @ErezCohenn , please note that first method returns the first object matched by the queryset, or None if there is no matching object. In addition ,  At this point we still don't have implementation for dogwner_homepage or daycare_object.
Furthermore , until now the render function include in the request only User information and not the entire object of Day-Care or Dog-Owner(which we need some fields from them for the navbar), so you can see context_processor.py that will add for every request our desirable fields and with this solution we get much shorther views .